conda/plugins/manager.py

```diff
@@ -36,11 +38,45 @@ def load_plugins(self, *plugins) -> list[str]:
                 plugin_names.append(plugin_name)
         return plugin_names
 
+    def get_registered_plugins(self, plugin_name: str) -> list:
+        """
+        Return all registered plugins for the given name.
+        """
+        if not plugin_name:
```

Comment:
If we do a check below to see if the hook attribute exists on the `self.hook` object, is this check really necessary? Even if the string value is empty it will still raise a nice exception and not blow up in a stack trace.

Furthermore, by being a positional argument, it's already assumed that we will pass in a value and because we use the `str` type hint, a string is strongly encouraged 😅 

I think we could ditch this check and make the code a little more legible. Definitely optional though.

Reply:
FWIW I added it to handle passing in an empty string, but that'll still be handled by the check for the hook below that. 👍